File: app/app/src/main/java/com/example/hakonsreader/api/interfaces/ReplyableRequest.java
package com.example.hakonsreader.api.interfaces;

import com.example.hakonsreader.api.model.RedditComment;

/**
 * Interface for request classes that offers requests for adding replies (reply to post, comment etc)
 *
 * <p>This interface is intended to be used with methods from {@link com.example.hakonsreader.api.RedditApi}
 * to use the same code for replying to different types of listings (comments or posts). </p>
 */
public interface ReplyableRequest {
    void reply(String text, OnResponse<RedditComment> onResponse, OnFailure onFailure);
}

File: app/app/src/main/java/com/example/hakonsreader/api/interfaces/ReplyableRequest.kt
package com.example.hakonsreader.api.interfaces

import com.example.hakonsreader.api.model.RedditComment
import com.example.hakonsreader.api.model.RedditListing
import com.example.hakonsreader.api.responses.ApiResponse

/**
 * Interface for a [RedditListing] that can be replied to
 */
interface ReplyableRequest {

    /**
     * Reply to a listing
     *
     * @param text The markdown text of the reply
     * @return An ApiResponse with the comment created
     */
    suspend fun reply(text: String) : ApiResponse<RedditComment>
}